wrappers/python/tests/layer/test_t5_attention.py

```diff
@@ -0,0 +1,325 @@
+    # Set input tensor dimensions
+    x_shape = [params.d_model, params.seq_len, params.n_batch]
+    x_basetile = [params.d_model_tile, params.seq_len_tile, params.n_batch_tile]
+    x_traits = TensorTraits(x_shape, x_basetile)
+    x_distr = [0] * x_traits.grid.nelems
+    x_type = dtype2nntile[dtype]
+    x_value = x_type(x_traits, x_distr, 0)
+    x_grad = x_type(x_traits, x_distr, 0)
+    X = TensorMoments(x_value, x_grad, grad_required=True)
+
+    # Generate random input data
+    gen = np.random.default_rng(42)
+    x_random = gen.standard_normal(x_shape, dtype=np.float32)
+    x_nntile = np.array(x_random, dtype=np.float32, order="F")
+    x_value.from_array(x_nntile)
+    x_torch = torch.Tensor(x_nntile.T)
+    x_torch.requires_grad_()
+
+    encoder_output_torch = None
+    if is_cross_attn:
+        encoder_output_traits = TensorTraits(x_shape, x_basetile)
+        encoder_output_distr = [0] * encoder_output_traits.grid.nelems
+        encoder_output_type = dtype2nntile[dtype]
+
+        encoder_output_value = encoder_output_type(
+            encoder_output_traits, encoder_output_distr, 0
+        )
+        encoder_output_grad = encoder_output_type(
+            encoder_output_traits, encoder_output_distr, 0
+        )
+
+        encoder_output_random = gen.standard_normal(x_shape, dtype=np.float32)
+        encoder_output_nntile = np.array(
+            encoder_output_random, dtype=np.float32, order="F"
+        )
+        encoder_output_value.from_array(encoder_output_nntile)
+        encoder_output_torch = torch.Tensor(encoder_output_nntile.T)
+        encoder_output_torch.requires_grad_()
+
+        encoder_output = TensorMoments(
+            encoder_output_value, encoder_output_grad, grad_required=True
+        )
+
+    # Initialize NNTile layer from PyTorch layer
+    if is_cross_attn:
+        nntile_layer, _ = T5Attention.from_torch(
+            torch_layer, X, None, nntile_config, 0, encoder_output=encoder_output
+        )
+    else:
+        nntile_layer, _ = T5Attention.from_torch(torch_layer, X, None, nntile_config, 0)
+    # nntile_layer.clear_gradients()
+
+    # Generate random gradient for backward pass
+    y_grad_random = gen.standard_normal(x_shape, dtype=np.float32)
+    y_grad_nntile = np.array(y_grad_random, dtype=np.float32, order="F")
+    nntile_layer.activations_output[0].grad.from_array(y_grad_nntile)
+    y_grad_torch = torch.Tensor(y_grad_nntile.T)
+
+    return torch_layer, nntile_layer, x_torch, y_grad_torch, encoder_output_torch
+
+
+@pytest.mark.parametrize(
+    "params",
+    [
+        pytest.param(single_tile, id="single_tile"),
+        pytest.param(multiple_tiles, id="multiple_tiles"),
+    ],
+)
+@pytest.mark.parametrize(
+    "dtype",
+    [
+        "fp32",
+        pytest.param("fp32_fast_tf32", marks=nocuda),
+        pytest.param("bf16", marks=nocuda),
+    ],
+)
+@pytest.mark.parametrize(
+    "is_cross_attn",
+    [
+        True,
+        False,
+    ],
+)
+class TestT5Attention:
```

Comment:
test_coercion function is missing, that checks from_torch and to_torch conversions like [here](https://github.com/nntile/nntile/blob/5b8493f2a3a7bb3c2ee7359a35949afa9135533c/wrappers/python/tests/layer/test_gpt2_attention.py#L144)